agent/src/beerocks/slave/tasks/task_pool.h

```diff
@@ -0,0 +1,60 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _TASK_POOL_H_
+#define _TASK_POOL_H_
+
+#include "task.h"
+#include <unordered_map>
+
+namespace beerocks {
+
+/**
+ * @brief The TaskPool is single instance tasks container which responsible to run each
+ * task and allow passing messages and events from the Agent to the task.
+ * 
+ */
+class TaskPool {
+public:
+    /**
+     * @brief Add new task to the task pool.
+     * 
+     * @param new_task Shared pointer to the task.
+     */
+    void add_task(std::shared_ptr<Task> new_task);
+
+    /**
+     * @brief Run all tasks on the pool, by calling each task work() function.
+     */
+    void run_tasks();
+
+    /**
+     * @brief Send an 'event' defined on a specific task 'task_type'. 
+     * 
+     * @param task_type Task type, defined on Task base class.
+     * @param event Event type, defined on the task itself.
+     */
+    void send_event(Task::eType task_type, uint8_t event);
```

Comment:
what happens if you have 2 tasks of the same type? task type isn't enough

Reply:
Not possible. This task pool support a single instance of a specific task.

Reply:
I don't think this is good.
the DCS, for example, has 2 task instances - one per radio.
we are currently limited to one-scan-at-a-time - and there isn't a requirement for simultaneous scans on multiple radios at the same time - but if that will change - the task class should not be hard limiting us like that.

an index with default value 0 is preferable, or some other key to allow multiple tasks of the same type.
(side-note - the current controller task_pool support task id for that - but you can suggest something else of course)
@kantera800

Reply:
You can manage DCS for different radio under the same task.
This is the new approach now. The unified agent will manage all fronthauls so the topology task and so on.

Reply:
ok